internal/provisioner/helm/controllers/bundledeployment_controller.go

```diff
@@ -223,6 +223,9 @@ func (r *BundleDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 				}))
 			return ctrl.Result{}, err
 		}
+		u.UpdateStatus(
+			updater.EnsureIncreaseGeneration(),
+		)
```

Comment:
We need to _always_ ensure the observed generation matches the generation. We should call this basically immediately after getting the BD, passing in the BD's `metadata.generation`